### DISCUSSION TITLE
Does HiGHS Support Custom Solving Algorithms?

### DISCUSSION THREAD

**Discussion by niconi19** [ERGO-Code/HiGHS#1776] [Ideas]
I am interested in testing some learning-based branching-variable-selection (BVS) algorithms, as well as presolving algorithms, for solving MIPs within HiGHS. Would it be possible to implement these customized algorithms into HiGHS for problem solving? Thanks!

**Comment by jajhall** [comment]
Inserting any new code into the MIP presolve or solve routines will be very difficult, so I would advise you not to try, even in your own fork. We certainly can't allow experimental ideas like this to be merged into the main HiGHS codebase without assurances of long-term support and being possible to switch them off if they compromise performance

**Reply by niconi19** [reply]
Thank you for the clarification. I appreciate your advice.
